app/models/user.rb

```diff
@@ -1,12 +1,41 @@
 class User < ApplicationRecord
   VALID_EMAIL_REGEX = Settings.valid_email_regex
+  attr_accessor :remember_token
 
-  before_save{email.downcase!}
+  before_save{self.email = email.downcase}
```

Comment:
```suggestion
  before_save{email.downcase!}
```
downcase! vs downcase khác gì nhau e ?

Reply:
downcase và downcase! dùng để in thường các ký tự trong chuỗi
Tuy nhiên downcase! có thể thay đổi giá trị của chuỗi ban đầu còn downcase thì không,
downcase! sẽ trả về nil nếu xâu ký tự không có gì thay đổi